src/stratigraphy/util/geometric_line_utilities.py

```diff
@@ -244,14 +244,15 @@ def _odr_regression(x: ArrayLike, y: ArrayLike) -> tuple:
     Args:
         x (ArrayLike): The x-coordinates of the data.
         y (ArrayLike): The y-coordinates of the data.
+        weights (ArrayLike): The weight for each data point.
 
     Returns:
         tuple: (phi, r), the best fit values for the line equation in normal form.
     """
-    x_mean = np.mean(x)
-    y_mean = np.mean(y)
-    nominator = -2 * np.sum((x - x_mean) * (y - y_mean))
-    denominator = np.sum((y - y_mean) ** 2 - (x - x_mean) ** 2)
+    x_mean = np.mean(np.dot(weights, x)) / np.sum(weights)
+    y_mean = np.mean(np.dot(weights, y)) / np.sum(weights)
+    nominator = -2 * np.sum(np.dot(weights, (x - x_mean) * (y - y_mean)))
+    denominator = np.sum(np.dot(weights, (y - y_mean) ** 2 - (x - x_mean) ** 2))
```

Comment:
Is this consistent?

I.e. if we are to use weights like [2, 2, 2, 2] would it lead to the same result as [1, 1, 1, 1]?

I am not sure if the weights would cancel.

I don't say it's wrong. I'm just confused. Let's discuss.

Reply:
For me, it would make sense to weight the differences `(x - x_mean)` and `(y - y_mean)`.

I.e. weight all the `(x_i - x_mean)` and `(y_i - y_mean)` with `weights_i`.

Thus, something like:

```
x_mean = np.mean(np.dot(weights, x)) / np.sum(weights)
y_mean = np.mean(np.dot(weights, y)) / np.sum(weights)
nominator = -2 * np.sum(np.dot(weights**2, (x - x_mean) * (y - y_mean)))
denominator = np.sum(np.dot(weights**2, (y - y_mean) ** 2 - (x - x_mean) ** 2))
```

Since the weights for x and y are the same, we can just square them.

Reply:
A better test would be something like: `[1, 1, 2, 2]` and `[2, 2, 4, 4]`.

Reply:
I have reverted this change, going back to using the simple weights, instead of the squared weights here.

I believe that this is mathematically more correct, though for our purposes, it does not change much. (At least in the case of perfectly parallel input lines, the resulting merged lines will be identical.)

Section "V.A." of https://www.researchgate.net/publication/272179120 explicitly gives formulas for weighted orthogonal distance regression, and these use the simple weights, not the squared weights. The mathematical derivation of these formulas also seems correct to me. Moreover, arguments resting on "dimensional analysis" can't really be used here, as the weights are unitless/dimensionless anyway, and stay the same even when taking a derivative with respect to r or phi (which actually occurs in the derivation of the formulas).

Reply:
And yes, using weights like [2, 2, 2, 2] leads to the same result as [1, 1, 1, 1]. Multiplying weights by a constant factor cancels out everywhere.